src/model/id.rs

```diff
@@ -1,30 +1,48 @@
 //! A collection of newtypes defining type-strong IDs.
 
 use std::fmt;
+use std::num::{NonZeroI64, NonZeroU64};
 
 use super::Timestamp;
 
 macro_rules! id_u64 {
     ($($name:ident;)*) => {
         $(
             impl $name {
+                /// Creates a new Id from a u64
+                ///
+                /// # Panics
+                /// Panics if the id is zero.
+                #[inline]
+                #[must_use]
+                pub fn new(id_as_u64: u64) -> Self {
+                    Self(NonZeroU64::new(id_as_u64).unwrap())
```

Comment:
There is no need to encode the type in the variable name.
```suggestion
                pub fn new(id: u64) -> Self {
                    Self(NonZeroU64::new(id).unwrap())
```